signer/src/keys.rs

```diff
@@ -51,6 +51,12 @@ impl Deref for PublicKey {
     }
 }
 
+impl From<&PublicKey> for PublicKey {
+    fn from(pk: &PublicKey) -> Self {
+        *pk
+    }
+}
+
```

Comment:
We can remove this, if we use a concrete type for `other` in `has_same_pubkeys`.

Reply:
[ef75378](https://github.com/stacks-sbtc/sbtc/pull/1602/commits/ef7537847ffe636587abe7e733399f34bf95adc7)